docs/rules/mockery.md

```diff
@@ -0,0 +1,64 @@
+# Mockery
```

Comment:
@h1alexbel You are using this term in many places in the code. Maybe we can add a link to the [description of this anti-pattern](https://stackoverflow.com/a/333686/10423604)? WDYT?